main.rb

```diff
@@ -0,0 +1,53 @@
+# this function accepts an array of numbers to be sorted
+# it sorts them and returns the sorted array
+def bubble_sort(arr)
+  sorted = false # a control for the sorting while loop
+
+  while (!sorted) do # keep iterating while sorted is false
+    operate = false
+    arr.length.times do |i|
+      current_number = arr[i]
+      next_number = arr[i + 1] || current_number
+
+      if current_number > next_number # compare the current and next number then swap is current is bigger than next
+        arr[i] = next_number
+        arr[i + 1] = current_number 
+        operate = true
+      end
+    end
+
+    if operate == false
+      sorted = true
+    end
+  end
+
+  puts "#{arr}"
+  return arr
+end
+
+bubble_sort([4,3,78,2,0,2]) # [0,2,2,3,4,78]
+bubble_sort([23, 8, 12, 0, 12, 7, 2, 4, 10, 1]) # [0, 1, 2, 4, 7, 8, 10, 12, 12, 23]
+
+def bubble_sort_by(arr)
+  sorted = false
+
+  while(!sorted) do
+    operation = false
+    arr.length.times do |i|
+      current_element = arr[i]
+      next_element = arr[i + 1] || current_element
+
+      if yield(current_element,next_element) > 0
+        arr[i] = next_element
+        arr[i + 1] = current_element
+        operation = true
+      end
+    end
+    break unless sorted
+  end
+  puts "#{arr}"
```

Comment:
Let's return `arr` instead of using `puts`

Reply:
Alright...noted